### ISSUE TITLE
jpeg-turbo 3.0.1 test failure on 12-arm build

### ISSUE THREAD

**Issue by chenrui333** [libjpeg-turbo/libjpeg-turbo#734]
👋 trying to build the latest release, but run into some build issue (something similar to https://github.com/libjpeg-turbo/libjpeg-turbo/issues/709). The error log is as below:

<details>
<summary>error build log</summary>

```
  The following tests FAILED:
  	246 - djpeg12-shared-3x2-float-prog-cmp (Failed)
  	541 - djpeg12-static-3x2-float-prog-cmp (Failed)
  Errors while running CTest
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/6807038616/job/18509256563?pr=153727
relates to Homebrew/homebrew-core#153727
This is the previous run, https://github.com/Homebrew/homebrew-core/actions/runs/6655988129/job/18087821345

some config

```
Clang: 14.0.0 build 1400
macOS: 12.6.3-arm64
Xcode: 14.2
```

**Comment by dcommander** [comment]
That is really weird.  Because of #709 and related issues, I rigorously tested all permutations of GCC v5-12 and Clang v5-14 on Linux/AArch64, GCC v4 on Linux/PPC, and GCC v8-10 and Clang v6-12 on Ubuntu/PPCLE with default and overridden `-ffp-contract` options.  That led to 47656a082091f9c9efda054674522513f4768c6c, which changes the expected MD5 sums for our floating point IDCT tests when using Clang with `FLOATTEST8=fp-contract` and `FLOATTEST12=fp-contract`, and 5b2beb4bc4f41dd9dd2a905cb931b8d5054d909b, which determines the appropriate default values of `FLOATTEST8` and `FLOATTEST12` for Arm and PowerPC builds based on the Clang version (recognizing that Clang v14 now enables floating point expression contraction by default.)  That fixed #709, which was reported against macOS 13 and Xcode 14.3/Clang v14.0.3.  However, from your build logs, it appears that Clang v14.0.0 in Xcode 14.2 does not enable floating point expression contraction by default, even though upstream Clang v14.0.0 does.  Perhaps Apple waited until Xcode 14.3 to merge the upstream change that turned on floating point expression contraction by default.

tl;dr: This issue will probably go away by upgrading to Xcode 14.3, but in the meantime, you should be able to get around it by passing `-DFLOATTEST12=no-fp-contract` to CMake when configuring the libjpeg-turbo build.  (Since `FLOATTEST8=fp-contract` and `FLOATTEST8=no-fp-contract` are equivalent when using Clang, there is no need to override `FLOATTEST8`.)

**Comment by dcommander** [comment]
NOTE: I would be willing to change the default `FLOATTEST*` values for Xcode 14.2 if (a) there is a reliable way for our build system to detect the Xcode version and (b) we can determine definitively that my hypothesis is correct (that Apple waited until Xcode 14.3 to merge the floating point expression contraction change.)
